src/logging/Logger.cpp

```diff
@@ -1,53 +1,158 @@
-#include "Logger.hpp"
 #include <boost/log/attributes/constant.hpp>
-#include <boost/log/attributes/mutable_constant.hpp>
+#include <boost/log/attributes/function.hpp>
 #include <boost/log/attributes/named_scope.hpp>
-#include <boost/log/core.hpp>
-#include <boost/log/expressions.hpp>
+#include <boost/log/attributes/timer.hpp>
+#include <boost/log/sources/severity_feature.hpp>
 #include <boost/log/trivial.hpp>
 #include <boost/log/utility/setup/common_attributes.hpp>
-#include <iosfwd>
 #include <utility>
+#include <utils/assertion.hpp>
+
+#include "logging/LogConfiguration.hpp"
+#include "logging/Logger.hpp"
 
 namespace precice {
 namespace logging {
 
+/// A feature that adds \ref LogLocation info to log attributes
+template <class BaseT>
+class precice_feature : public BaseT {
+public:
+  using char_type       = typename BaseT::char_type;
+  using threading_model = typename BaseT::threading_model;
+
+  precice_feature()                             = default;
+  precice_feature(const precice_feature &other) = default;
```

Comment:
Do you intentionally leave out other SMFs? What about a destructor?

Reply:
Good point. I could remove both of them, really.